### PR TITLE
Fix for gravitee-io/issues#5301 - Bump dependency versions

### PR DESCRIPTION
**Pull request by fuzzylimes**
* Changes to dependencies
    * Bumped the versions of Gravitee packages
    * removed the unneeded referenced to junit (handled by parent)
    * fixed issue with mockito reference
* Fixed policy assembly issues
    * Added README.adoc to assembly list
    * Added modified schema folder (`src/main/resources/schemas`) to fileSet
        * Pulls in user modified versions of schema, if available
* Updates to documentation
    * fixed some minor formatting issues with tables and links
    * Added some additional details as to how the JSON schemas work
    * added a short example showing additional, optional, schema properties

### CODE REVIEW

**Comment by fuzzylimes** [comment]
Fix for gravitee-io/issues#5301

File: pom.xml [hunk 1]
```diff
@@ -21,7 +21,7 @@
 
     <groupId>io.gravitee.maven.archetypes</groupId>
     <artifactId>gravitee-policy-maven-archetype</artifactId>
-    <version>1.9.0-SNAPSHOT</version>
+    <version>1.9.0</version>
```
Comment (aelamrani) [inline pom.xml]: The version has to remain snapshot until released

File: src/main/resources/archetype-resources/src/assembly/policy-assembly.xml [hunk 2]
```diff
@@ -37,6 +37,19 @@
...
+        </fileSet>
+
+        <!-- Include the Policy documentation -->
+        <fileSet>
+            <directory>${basedir}</directory>
+            <includes>
+                <include>README.adoc</include>
+            </includes>
+            <outputDirectory>docs</outputDirectory>
+        </fileSet>
```
Comment (aelamrani) [inline src/main/resources/archetype-resources/src/assembly/policy-assembly.xml:52]: What about adding a fileset for the icon as well? 

See: https://github.com/gravitee-io/gravitee-policy-mock/blob/master/src/assembly/policy-assembly.xml#L49

And changing the plugin.properties file to add the category and icon?

See: https://github.com/gravitee-io/gravitee-policy-mock/blob/master/src/main/resources/plugin.properties#L7

**Review by aelamrani — CHANGES_REQUESTED** [review]
Thanks for this contribution, a few comments here.

File: src/main/resources/archetype-resources/src/assembly/policy-assembly.xml (hunk 2 above)
Comment (fuzzylimes) [reply src/main/resources/archetype-resources/src/assembly/policy-assembly.xml:52]: Including fileset for icon and a default category is no problem. I have those updates working.

However, adding icon to `plugin.properties` will cause a `An error occurs while trying to get plugin icon for plugin beta-archetype-policy` error unless some icon file is included. I have tried with:

```
# dummy file
icon=icon.svg
# no file
icon=
```

How would you like me to handle this? A few options could be:

1. Include the mock icon from the `gravitee-policy-mock` project you linked, so it's included with generation
2. Update the readme to include an `adding custom icon` section.

Comment (aelamrani) [reply src/main/resources/archetype-resources/src/assembly/policy-assembly.xml:52]: Ah yes, you are right.
I would go for 2, and maybe we can just remove it from plugin.properties file to avoid failures (not a mandatory value).

File: README.adoc [hunk 3]
```diff
@@ -21,7 +21,7 @@ Suppose you want to create a Policy that control if request contains the ``X-Foo
 mvn archetype:generate\
  -DarchetypeGroupId=io.gravitee.maven.archetypes\
  -DarchetypeArtifactId=gravitee-policy-maven-archetype\
- -DarchetypeVersion=1.9.0\
+ -DarchetypeVersion=1.9.0-SNAPSHOT\
```
Comment (aelamrani) [inline README.adoc]: Can you revert this change? The version will be 1.9.0 once published.

File: pom.xml (hunk 1 above)
Comment (fuzzylimes) [reply pom.xml]: Fixed

File: src/main/resources/archetype-resources/src/assembly/policy-assembly.xml (hunk 2 above)
Comment (fuzzylimes) [reply src/main/resources/archetype-resources/src/assembly/policy-assembly.xml:52]: Sounds good. I debated leaving it in but commented out (`#icon`), but have removed it and added some additional documentation.

File: README.adoc (hunk 3 above)
Comment (fuzzylimes) [reply README.adoc]: Just to confirm - You want the version in the documentation to remain `1.9.0`, but pom to stay snapshot?

**Comment by aelamrani** [comment]
LGTM, may I ask you to squash your commits to a single one?

File: README.adoc [hunk 4]
```diff
@@ -201,20 +231,28 @@ As said, a Policy is a kind of Gravitee Plugin.
...
+| `class` | The main Policy class     | Path to the generated class file
+| `type` | The type of Gravitee Plugin     | `policy`
+| `category` | The category the generated plugin will appear under on the designer | `others`
+| `icon` | Custom icon file name | N/A (optional parameter)
 |===
 
 > A Policy is enabled when declared into the API definition. To do so, the Policy identifier is used to, as its name indicate, identify the Policy. Thus, **be ware to correctly choose the Policy identifier** from the beginning. It could be hard to rename it later if there are many of API definitions linked to it.
 
+==== Custom Icons
+In order to add a custom icon to your plugin, as displayed on the API Designer, you should:
```
Comment (aelamrani) [inline README.adoc]: API Designer => Design Studio

**Comment by fuzzylimes** [comment]
@aelamrani Fixed your comment about Design Studio and squashed everything.
